dist/ExtUtils-ParseXS/lib/ExtUtils/ParseXS/Node.pm

```diff
@@ -39,23 +39,30 @@ my $close_brace = '}';
 # BEGIN. (Created as a lexical sub ref to make it easily accessible to
 # all subclasses in this file.)
 #
-# Arg 1 is name of parent, without E::PXS::Node:: prefix.
-# Remaining args are the names of fields. It also inherits the fields
+# Arg 1 and 2 should be
+#         'parent' => Foo
+# where the node's class should be ExtUtils::ParseXS::Node::Foo prefix.
+# An empty string implies the parent is E::PXS::Node .
+#
+# Any remaining args are the names of fields. It also inherits the fields
 # of its parent.
 
 my $USING_FIELDS;
 
 my $build_subclass;
 BEGIN {
     $build_subclass = sub {
-        my $parent = shift;
+        my ($key, $parent, @fields) = @_;
+
+        die "Internal error: key '$key' not 'parent'\n"
+            unless $key eq 'parent';
```

Comment:
Right now we have a bunch of `parent => ''`, rather than forcing an empty parameter that implies a bare `...::Node` base class, maybe `parent` could be optional (perhaps renamed to `-parent` to avoid `parent` being the only bare identifier with special behaviour).

So all the places that now do `$build_subclass->(parent => '', fields...)` would be `$build_subclass->(fields...)`.

Calls that need a non-bare-::Node base class would be $build_subclass->(-parent=> 'parentname', fields...)`.

Reply:
Yeah, that sounds sensible. I'll probably do something along those lines. (NB: I'm not intending to make any changes until after you've reviewed the entirety of the PR in its current state.)